e2e/testcases/otel_collector_test.go

```diff
@@ -212,6 +216,157 @@ func TestOtelCollectorDeployment(t *testing.T) {
 	}
 }
 
+// TestOtelCollectorSampleConfigurations validates that metrics reporting works for
+// Google Cloud Monitoring using the sample custom configurations.
+//
+// Requirements:
+// - node identity:
+//   - node GSA with roles/monitoring.metricWriter IAM
+//
+// - workload identity:
+//   - e2e-test-metric-writer GSA with roles/monitoring.metricWriter IAM
+//   - roles/iam.workloadIdentityUser on config-management-monitoring/default for e2e-test-metric-writer
```

Comment:
We don't need this anymore with BYOID, right?

Reply:
Good catch. Tweaking the setup now, the tests are passing on failure scenarios, still working on it.